rfcs/0060-content-addressed-paths.md

```diff
@@ -0,0 +1,281 @@
+---
+feature: Simple content-adressed store paths
+start-date: 2019-08-14
+author: Théophane Hufschmitt
+co-authors: (find a buddy later to help our with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+
+[summary]: #summary
+
+Add some basic but simple support for content-adressed store paths to Nix.
+
+We plan here to give the possibility to mark certain store paths as
+content-adressed (ca), while keeping the other dependency-adressed as they are
+now (modulo some mandatory drv rewriting before the build, see below)
+
+By making this opt-in, we can impose arbitrary limitations to the paths that
+are allowed to be ca to avoid some tricky issues that can arise with
+content-adressability.
+In particular, we restrict ourselves to paths without any non-textual
+self-reference (_i.e_ a self-reference hidden inside a zip file) and known to
+be deterministic (for caching reasons, see [caching]).
+That way we don't have to worry about the fact that hash-rewriting is only an
+approximation nor by the semantics of the distribution of non-deterministic
+paths, **but** we also leave the option to lift these restrictions later.
```

Comment:
This paragraph is quite complex, maybe break it up in to a simpler sentence with a bulleted list?

```md

By making this opt-in, we can impose arbitrary limitations and avoid some tricky issues that can arise with content-adressability. In particular, we restrict ourselves to:

* paths without any non-textual
self-reference (_i.e_ a self-reference hidden inside a zip file)
* builds known to be deterministic for caching, see: [caching]

These limitations mean we  don't have to worry about the fact that hash-rewriting is only an approximation, nor do we need to worry about the semantics of the distribution of non-deterministic paths.

We also leave the option to lift these restrictions later. 
```